interview/algorithm/sort.py

```diff
@@ -52,6 +52,24 @@ def partition(array, low, high):
     return pb - 1
 
 
+def bubblesort(array):
```

Comment:
Another way to do this is placing the element in the right location with lesser swaps, but won't be able to early stop, so it's a trade-off
```
for i in range(n):
    for j in range(i+1, n):
        if array[i] > array[j]:
            swap(array, i, j)
```

Reply:
Yes. With the current implementation and the early termination it will run in O(N) for arrays already sorted.